src/algebraic_geometry/presheafed_space/gluing.lean

```diff
@@ -0,0 +1,434 @@
+      (D.V (j, k)).presheaf.map (eq_to_hom _),
+    dsimp only [functor.op, opens.map, unop_op],
+    congr' 2,
+    rw set.preimage_preimage,
+    change (D.f j k ≫ 𝖣 .ι j).base ⁻¹' _ = _,
+    congr' 3,
+    exact colimit.w 𝖣 .diagram.multispan (walking_multispan.hom.fst (j, k)) },
+  { exact D.opens_image_preimage_map i j U }
+end
+
+/-- (Implementation) The natural map `Γ(𝒪_{U_i}, U) ⟶ Γ(𝒪_X, 𝖣.ι i '' U)`.
+This forms the inverse of `(𝖣.ι i).c.app (op U)`. -/
+def ι_inv_app (i : D.J) (U : opens (D.U i).carrier) :
+  (D.U i).presheaf.obj (op U) ⟶ limit (componentwise_diagram 𝖣 .diagram.multispan
+    ((D.ι_open_embedding i).is_open_map.functor.obj U)) :=
+limit.lift (componentwise_diagram 𝖣 .diagram.multispan
+    ((D.ι_open_embedding i).is_open_map.functor.obj U))
+{ X := (D.U i).presheaf.obj (op U),
+  π := { app := λ j, D.ι_inv_app_π_app i U (unop j),
+  naturality' := λ X Y f', begin
+    induction X using opposite.rec,
+    induction Y using opposite.rec,
+    let f : Y ⟶ X := f'.unop, have : f' = f.op := rfl, clear_value f, subst this,
+    rcases f with (_|⟨j,k⟩|⟨j,k⟩),
+    { erw [category.id_comp, category_theory.functor.map_id],
+      rw category.comp_id },
+    { erw category.id_comp, congr' 1 },
+    erw category.id_comp,
+    change D.opens_image_preimage_map i j U ≫
+      (D.f j k).c.app _ ≫
+        (D.V (j, k)).presheaf.map (eq_to_hom _) =
+          D.opens_image_preimage_map _ _ _ ≫
+            ((D.f k j).c.app _ ≫ (D.t j k).c.app _) ≫
+              (D.V (j, k)).presheaf.map (eq_to_hom _),
+    erw opens_image_preimage_map_app_assoc,
+    simp_rw category.assoc,
+    erw [opens_image_preimage_map_app_assoc, (D.t j k).c.naturality_assoc],
+    rw snd_inv_app_t_app_assoc,
+    erw ← PresheafedSpace.comp_c_app_assoc,
+    have : D.t' j k i ≫ pullback.fst ≫ D.t k i ≫ 𝖣 .f i k =
+      (pullback_symmetry _ _).hom ≫ pullback.fst ≫ D.t j i ≫ D.f i j,
+    { rw [← 𝖣 .t_fac_assoc, 𝖣 .t'_comp_eq_pullback_symmetry_assoc,
+        pullback_symmetry_hom_comp_snd_assoc, pullback.condition, 𝖣 .t_fac_assoc] },
+    rw congr_app this,
+    erw PresheafedSpace.comp_c_app_assoc (pullback_symmetry _ _).hom,
+    simp_rw category.assoc,
+    congr' 1,
+    rw ← is_iso.eq_inv_comp,
+    erw is_open_immersion.inv_inv_app,
+    simp_rw category.assoc,
+    erw [nat_trans.naturality_assoc, ← PresheafedSpace.comp_c_app_assoc,
+      congr_app (pullback_symmetry_hom_comp_snd _ _)],
+    simp_rw category.assoc,
+    erw [is_open_immersion.inv_naturality_assoc, is_open_immersion.inv_naturality_assoc,
+      is_open_immersion.inv_naturality_assoc, is_open_immersion.app_inv_app_assoc],
+    repeat { erw ← (D.V (j, k)).presheaf.map_comp },
+    congr,
+  end } }
+
+lemma ι_inv_app_π (i : D.J) (U : opens (D.U i).carrier) :
+  ∃ eq, D.ι_inv_app i U ≫ limit.π (componentwise_diagram 𝖣 .diagram.multispan
+    ((D.ι_open_embedding i).is_open_map.functor.obj U)) (op (walking_multispan.right i)) =
+    (D.U i).presheaf.map (eq_to_hom eq) :=
```

Comment:
I've added an abbreviation for that and hopefully It looks better now.
I don't have any idea to shorten the statements though. It would probably be clearer if I can somehow include [this diagram](https://q.uiver.app/?q=WzAsMTMsWzIsMiwiVV9pIl0sWzEsNCwiVV9qIl0sWzMsNCwiVV9rIl0sWzEsNSwiVl97amt9Il0sWzIsMywiWCJdLFswLDQsIlZfe2ppfSJdLFswLDUsIlZfe2prfVxcdGltZXNfe1Vfan1WX3tqaX0iXSxbMSwxLCJWX3tpan0iXSxbMywxLCJWX3tpa30iXSxbMiwwLCJWX3tpan1cXHRpbWVzX3tVX2l9Vl97aWt9Il0sWzMsNSwiVl97a2p9Il0sWzQsNSwiVl97a2l9XFx0aW1lc197VV9rfVZfe2tqfSJdLFs0LDQsIlZfe2tpfSJdLFswLDRdLFsxLDRdLFsyLDRdLFs1LDFdLFszLDFdLFs2LDVdLFs2LDNdLFs3LDBdLFs4LDBdLFs5LDddLFs5LDhdLFsxMiwyXSxbMTEsMTJdLFsxMSwxMF0sWzEwLDJdLFs5LDExLCIiLDAseyJvZmZzZXQiOi01LCJjdXJ2ZSI6LTV9XSxbNiw5LCIiLDAseyJvZmZzZXQiOi01LCJjdXJ2ZSI6LTV9XSxbMTEsNiwiIiwwLHsiY3VydmUiOi01fV0sWzEwLDNdLFs1LDddLFs4LDEyXV0=) in the file, but I'm not that of an ASCII art master.

Reply:
@erdOne Maybe just include the link to that diagram. Then people can easily open it in a web browser. I think that would be helpful.